Automattic/wp-calypso
# Jetpack Manage: Upsells - secondary checkout support for LicenseLightbox including a SingleSiteUpsellLightbox component

Related to https://github.com/Automattic/jetpack-manage/issues/216
Also related to https://github.com/Automattic/jetpack-manage/issues/217

## Proposed Changes

* This PR allows for a new secondary checkout content area within the LicenseLightbox component. When the prop to allow this area to show exists, an 'Or' divider is added.
* This PR also creates a new 'SingleSiteUpsellLightbox' component, which is populated with upsell information and buttons, as per the design specs: p1HpG7-pC8-p2
* This PR also changes the wording showing above the main sidebar area in the License Lightbox - from 'Payment Plan' to 'Manage Jetpack License' (and changes the relevant components / div class names). See p1HpG7-pC8-p2#comment-67355


## Testing Instructions

* Check out this branch, and make sure your partner type is agency.

Testing the new content area:
* To test this new secondary sidebar content area in the License Lightbox component, set `showSecondaryContent` to true in `license-lightbox/index.tsx'.
* You can then visit Jetpack Manage > Sites and clicking the info button in the backup or scan table headers. The popup should show an 'Or' divider.

To test the new SingleSiteUpsellLightbox component:
* To test this out, you'll first need to call the new component. I tested this in a somewhat hacky way - click on the expandable content below for the code snippets.
 
<details>
<summary>To add to the `agency-dashboard/sites-overview/site-status-content/site-boost-column/boost-license-info-modal.tsx` file:</summary>

At the top of the file, add:
```
import SingleSellUpsellLightbox from 'calypso/jetpack-cloud/sections/partner-portal/license-lightbox/single-site-upsell-lightbox';
import { useSelector } from 'calypso/state';
import { getCurrentPartner } from 'calypso/state/partner-portal/partner/selectors';
```

Add this to just before the return in BoostLicenseInfoModal:
```
const { products } = useContext( DashboardDataContext );
const currentLicenseProduct = products.find(
	( product ) => product.slug === 'jetpack-backup-t1'
);
const partner = useSelector( getCurrentPartner );
const partnerCanIssueLicense = Boolean( partner?.can_issue_licenses );
```

and replace the `return ( <LicenseInfoModal....` content with:
```
return (
	<>
		{ currentLicenseProduct && (
			<SingleSellUpsellLightbox
				currentProduct={ currentLicenseProduct }
				onClose={ onClose }
				productSlug="jetpack_backup_t1_yearly"
				partnerCanIssueLicense={ partnerCanIssueLicense }
				siteId={ siteId }
			/>
		) }
	</>
);
```


</details>

 
* Then, from the same Jetpack Manage > Sites page as above, click on 'Get Score' from the Boost column, for a site that has this available (you may need to add and connect a new Jurassic Ninja site to see this).
* The design should match the design shared here: p1HpG7-pC8-p2
* Clicking 'issue license' should take you to the issue license page, clicking 'Purchase via Jetpack.com' should take you to the WPcom checkout (and include a URL redirect to send you back to the page you were on when done), and clicking any of those two buttons plus the 'learn more' links should fire tracks events.


<img width="1064" alt="Screenshot 2024-01-11 at 20 08 00" src="https://github.com/Automattic/wp-calypso/assets/16754605/ebc0ef80-4f48-4566-800b-b7db0184fa89">

## Pre-merge Checklist
